quarkus-cli/src/test/java/io/quarkus/ts/quarkus/cli/QuarkusCliCreateJvmApplicationIT.java

```diff
@@ -188,13 +212,13 @@ private void assertInstalledExtensions(QuarkusCliRestService app, String... expe
                 expectedExtension + " not found in " + extensions));
     }
 
-    private void assertExpectedJavaVersion(File pomFile) {
+    private void assertExpectedJavaVersion(File pomFile, String expectedJavaVersion) {
         MavenXpp3Reader reader = new MavenXpp3Reader();
-        String javaVersion = getSystemJavaVersion();
         try {
             Model model = reader.read(new FileReader(pomFile));
-            Assertions.assertEquals(model.getProperties().get("maven.compiler.release"), javaVersion,
-                    "Unexpected Java version. Java support tool should detect host Java version");
+            Assertions.assertEquals(model.getProperties().get("maven.compiler.release"), expectedJavaVersion,
+                    "Unexpected Java version. Java support tool should detect host Java version or use " +
+                            "the provided one by --java argument");
         } catch (IOException | XmlPullParserException e) {
```

Comment:
Also check generated Dockerfile.jvm. This is needed for good OpenShift experience.